pinot-common/src/main/java/org/apache/pinot/common/metrics/BrokerMeter.java

```diff
@@ -18,195 +18,134 @@
  */
 package org.apache.pinot.common.metrics;
 
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 import org.apache.pinot.common.Utils;
-
+import org.apache.pinot.spi.exception.QueryErrorCode;
 
 /**
- * Enumeration containing all the metrics exposed by the Pinot broker.
+ * Class containing all the metrics exposed by the Pinot broker.
```

Comment:
Why is the change from enum to class required ?

Reply:
This is a big structural change in terms of dev u/x. Metrics for other components as well as in Startree follow the current pattern. So this will become an odd one out. Are there other ways to achieve what you need ?

Reply:
I couldn't figure out another way apart from above mentioned. tried creating a seperate class but AbstractMetrics restricts that. and didn't want to add that option of manual mistake if someone adds an error code but misses adding a metric. 

Yea the class structure changes, but the usage of the class remains same. Clients still get to use same variables as earlier without any changes. and adding new metrics is also same effort as before by just creaing a new variable.

Reply:
I have validated the final metric names via comparing the bean names from this branch and master and doing a text diff

also compared unit and global flags are same for all variables manually.

Reply:
Is another option not to use this class ? Since you are generating metrics dynamically, create a separate class for that ? That seems like a clean separation. 
wrt to dev u/x adding a metric is different. 

One more thing - the javadocs have been removed.

Reply:
tried creating a seperate class but AbstractMetrics restricts that. 

thanks for the javadocs catch. I missed seeing it got removed

Reply:
The current metric system is very restrictive. We need to be able to extend it in the future. I don't love this change, but I think it is necessary.

Anyway, we are not consuming this class as an enum (ie, applying switches on values of this type or using them as enums in binary representation) so it shouldn't affect us that much.